src/sql/src/plan/typeconv.rs

```diff
@@ -48,45 +48,92 @@ impl From<UnaryFunc> for CastOp {
 
 impl CastOp {
     /// Generates the [`ScalarExpr`] to cast between [`ScalarType`]s.
-    pub fn gen_expr(&self, ecx: &ExprContext, e: ScalarExpr, cast_to: CastTo) -> ScalarExpr {
+    pub fn gen_expr(
+        &self,
+        ecx: &ExprContext,
+        e: ScalarExpr,
+        cast_to: CastTo,
+    ) -> Result<ScalarExpr, failure::Error> {
         match self {
-            CastOp::U(u) => e.call_unary(u.clone()),
+            CastOp::U(u) => Ok(e.call_unary(u.clone())),
             CastOp::F(f) => f(ecx, e, cast_to),
         }
     }
 }
 
 // Used when the [`ScalarExpr`] is already of the desired [`ScalarType`].
-fn noop_cast(_: &ExprContext, e: ScalarExpr, _: CastTo) -> ScalarExpr {
-    e
+fn noop_cast(_: &ExprContext, e: ScalarExpr, _: CastTo) -> Result<ScalarExpr, failure::Error> {
+    Ok(e)
 }
 
 // Cast `e` to `String`, and then to `Jsonb`.
-fn to_jsonb_any_string_cast(ecx: &ExprContext, e: ScalarExpr, _: CastTo) -> ScalarExpr {
+fn to_jsonb_any_string_cast(
+    ecx: &ExprContext,
+    e: ScalarExpr,
+    _: CastTo,
+) -> Result<ScalarExpr, failure::Error> {
     let s = ecx.scalar_type(&e);
     let to = CastTo::Explicit(ScalarType::String);
 
     let cast_op = get_cast(&s, &to).unwrap();
 
-    cast_op
-        .gen_expr(ecx, e, to)
-        .call_unary(UnaryFunc::CastJsonbOrNullToJsonb)
+    Ok(cast_op
+        .gen_expr(ecx, e, to)?
+        .call_unary(UnaryFunc::CastJsonbOrNullToJsonb))
 }
 
 // Cast `e` to `Float64`, and then to `Jsonb`.
-fn to_jsonb_any_f64_cast(ecx: &ExprContext, e: ScalarExpr, _: CastTo) -> ScalarExpr {
+fn to_jsonb_any_f64_cast(
+    ecx: &ExprContext,
+    e: ScalarExpr,
+    _: CastTo,
+) -> Result<ScalarExpr, failure::Error> {
     let s = ecx.scalar_type(&e);
     let to = CastTo::Explicit(ScalarType::Float64);
 
     let cast_op = get_cast(&s, &to).unwrap();
 
-    cast_op
-        .gen_expr(ecx, e, to)
-        .call_unary(UnaryFunc::CastJsonbOrNullToJsonb)
+    Ok(cast_op
+        .gen_expr(ecx, e, to)?
+        .call_unary(UnaryFunc::CastJsonbOrNullToJsonb))
+}
+
+fn to_jsonb_any_record_cast(
+    ecx: &ExprContext,
+    e: ScalarExpr,
+    _: CastTo,
+) -> Result<ScalarExpr, failure::Error> {
+    let fields = match ecx.scalar_type(&e) {
+        ScalarType::Record { fields } => fields,
+        _ => unreachable!(),
+    };
+
+    let mut exprs = vec![];
+    for (i, (name, _ty)) in fields.iter().enumerate() {
+        exprs.push(ScalarExpr::literal(
+            Datum::String(name.as_str()),
+            ColumnType::new(ScalarType::String),
+        ));
+        exprs.push(plan_cast(
+            "foo",
```

Comment:
I think this should be `to_jsonb` for `plan_cast`'s error message?

Reply:
D'oh, thanks!